SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1964,7 +1961,24 @@ ERPCResult USpatialReceiver::ApplyRPCInternal(UObject* TargetObject, UFunction*
 			UE_LOG(LogSpatialReceiver, Warning, TEXT("Executed RPC %s::%s with unresolved references (%s) after %.3f seconds of queueing.  Owner name: %s"), *GetNameSafe(TargetObject), *GetNameSafe(Function), *UnresolvedEntityIds, TimeQueued, *GetNameSafe(TargetObject->GetOuter()));
 		}
 
-		Result = ERPCResult::Success;
+		// Get the RPC target Actor.
+		AActor* Actor = TargetObject->IsA<AActor>() ? Cast<AActor>(TargetObject) : TargetObject->GetTypedOuter<AActor>();
+		ERPCType RPCType = PendingRPCParams.Type;
+
+		if (Actor->Role == ROLE_SimulatedProxy &&
+			(RPCType == ERPCType::ServerReliable ||
+			 RPCType == ERPCType::ServerUnreliable))
+		{
+			ErrorInfo.ErrorCode = ERPCResult::NoAuthority;
+			ErrorInfo.QueueProcessResult = ERPCQueueProcessResult::DropEntireQueue;
+			UE_LOG(LogSpatialReceiver, Warning, TEXT("Trying to execute Server RPC %s::%s without authority over the target actor. RPCs for this actor will be dropped."), *GetNameSafe(TargetObject), *GetNameSafe(Function));
```

Comment:
Do we drop server RPCs invoked on a non auth server (I know on the auth server we just invoke locally)

Reply:
This is defo a warning. This could happen if we have authority over an actor upon receiving the RPC. The RPC can not be processed because we are waiting on an unresolved argument in a previous RPC. We then lose authority over the RPC actor. 

So yes we are dropping all RPC in the queue for non auth actors. 

"can we do the string interpolation" Sorry I am not sure what you mean here?

Reply:
In the case you just described it shouldn't be warning though as the user has done nothing wrong and everything will work. The next work will pick up the rpc correctly. This is just some internal book keeping at that point.
Might be easier to discuss offline.

Reply:
Ahh yes, you are right. There is no need to warn.